### PR TITLE
Runner improvements

### PR DESCRIPTION
**Pull request by veruu**
Fix some pylint errors and bug with infinite jobs resubmitting

### CODE REVIEW

File: skt/runner.py [hunk 1]
```diff
@@ -149,9 +149,9 @@ def getconsolelog(self, jobid=None):
             jobid = self.lastsubmitted
 
         root = self.getresultstree(jobid, True)
-        el = root.find("recipeSet/recipe/logs/log[@name='console.log']")
-        if el is not None:
-            url = el.attrib.get("href")
+        console = root.find("recipeSet/recipe/logs/log[@name='console.log']")
+        if console is not None:
+            url = console.attrib.get("href")
```
Comment (spbnick) [inline skt/runner.py]: Do you think this would be better to name `console_log` instead?

File: skt/runner.py [hunk 2]
```diff
@@ -325,12 +325,17 @@ def getresults(self, jobid=None):
...
         if samehost:
-            hreq = tmp.find("hostRequires")
             hostname = etree.Element("hostname")
             hostname.set("op", "=")
             hostname.set("value", tmp.attrib.get("system"))
             hreq.append(hostname)
+        else:
+            hostname = hreq.find('hostname')
+            if hostname is not None:
+                hreq.remove(hostname)
```
Comment (spbnick) [inline skt/runner.py]: Hmm, shouldn't we simply remove the `hostname` requirement every time, and only add it back when `samehost` is True?

Comment (veruu) [reply skt/runner.py]: I don't understand how you mean it. If the `samehost` is true, we keep the hostname and if not, we check if there is a hostname specified in the recipe and remove it if there is one. Can you elaborate how you imagine the solution?

Comment (spbnick) [reply skt/runner.py]: Like this:

        hreq = tmp.find("hostRequires")
        hostname = hreq.find('hostname')
        if hostname is not None:
            hreq.remove(hostname)
        if samehost:
            hostname = etree.Element("hostname")
            hostname.set("op", "=")
            hostname.set("value", tmp.attrib.get("system"))
            hreq.append(hostname)

This way we make sure there's no `hostname` element when `samehost` is `False`, and there's only one `hostname` element, when it's `True` (that is unless there was more than one originally, which we can also handle, if needed).

Comment (veruu) [reply skt/runner.py]: Thanks for the explanation! Yes, that makes sense. I'll update the pull.

**Comment by veruu** [comment]
@spbnick updated, thanks for the idea again!

**Pull request merged**